### DISCUSSION TITLE
m-api for Home Assistant

### DISCUSSION THREAD

**Discussion by niro1987** [persandstrom/python-verisure#138] [General]
Hi,

Thanks for opening up the discussion page.
I've been a happy user of Home Assistant for a few years now, and I've recently had a Verisure alarm system installed.

I was verry happy to see that there is indeed an integration for Verisure in Home Assistant, supported by the great work of @persandstrom! Thank you so much!

The one thing that isn't working, is the Danalock V3. or at least, not in the master branch.

I've been looking at the Home Assistant integration, it takes the data from the `get_overview` function, and created entities from it. The `lock` entities are created from entries in the `doorLockStatusList`, this list does not include the Danalock smartlocks.

The **m-api** branch does have support for the danalock, but ..
> it is not compatible with home assistant

I've opened this post to discuss what is needed to get all features up-and-running.

**Comment by niro1987** [comment]
@persandstrom Is there a specific formatter that you've used in your python code? I'm trying to keep the code style the same, but my vscode is formatting it differently.

**Comment by persandstrom** [comment]
No, I have not been using any formatter. Do you plan to make an Adapter API on the python-verisure side, or to create an HA integration specialized for m-api branch?

**Comment by niro1987** [comment]
I'm looking to create an HA integration for the m-api branch. I feel it would be best to release an updated package based on the m-api branch, if and when the m-api branch has all the same features as the master branch (based on graphql requests).

**Comment by niro1987** [comment]
I think we need to create a package on pypi in order to use it for a HA integration. Is the m-api branch also published to pypi?

**Reply by persandstrom** [reply]
I think you are correct. It is not published yet. I'll publish it when MFA is added

**Comment by persandstrom** [comment]
The m-api branch is renamed to "version-2" and is now the main branch. A release based on version-2 is uploaded to pypi

**Reply by niro1987** [reply]
Congratulations! I'll start looking into the Home Assistant integration.

**Reply by persandstrom** [reply]
Thank you for all support @niro1987 ! Added the first commit for m-api branch "Wed Oct 28 18:35:34 2020" and it would probably stayed on ice a lot longer without your contributions

**Comment by persandstrom** [comment]
Hi @niro1987 !
How is it going with the move to m-api for HomeAssistant?

**Reply by niro1987** [reply]
Hi @persandstrom 
I have a working component update here: [https://github.com/niro1987/homeassistant_core/tree/verisure](https://github.com/niro1987/homeassistant_core/tree/verisure)

You can test it by creating a fork (make sure to also fork the verisure branch) and running a devcontainer as described in the [development docs](https://developers.home-assistant.io/docs/development_environment#developing-with-visual-studio-code--devcontainer).

**Reply by persandstrom** [reply]
Good that you are close to a solution. I believe that Verisure has closed down the old API now.

**Reply by niro1987** [reply]
PR is opened

**Reply by persandstrom** [reply]
Perfekt!

**Reply by niro1987** [reply]
Mind helping out with some of this feedback?

https://github.com/home-assistant/core/pull/89318

**Reply by persandstrom** [reply]
I responded to some, any other in particular you need support with?

**Reply by niro1987** [reply]
Mind taking a look at https://github.com/home-assistant/core/pull/89318#issuecomment-1475188875 and https://github.com/home-assistant/core/pull/89318#discussion_r1142401784

**Comment by persandstrom** [comment]
Thank you for fixing the HA integration @niro1987 !

**Reply by niro1987** [reply]
Thanks for the help!
